compiler-rt/src/alu/add.cairo

```diff
@@ -0,0 +1,36 @@
+pub mod add_i8;
+
+use crate::utils::assert_fits_in_type;
+use core::num::traits::{BitSize, WrappingAdd};
+
+// Perform the `add` operation.
+//
+// This is a generic implementation for every data type. Its specialized versions
+// are defined and tested in add/add_i<type>.cairo files.
+fn add<
+    T,
+    // The trait bounds are chosen so that:
+    //
+    // - BitSize<T>: we can determine the length of the data type in bits,
+    // - TryInto<u128, T>, Into<T, i128>: we can convert the type to i128 and from u128. This is
+    // necessary, because we can convert i128->u128, but we cannot convert signed integers -> u128.
+    // - Destruct<T>: the type can be dropped as the result of the downcasting check.
+    // - WrappingAdd<T>: so we can use T.wrapping_add().
+    //
+    // Overall these trait bounds allow any unsigned integer to be used as the concrete type.
+    impl TBitSize: BitSize<T>,
+    impl TTryInto: TryInto<u128, T>,
+    impl TInto: Into<T, u128>,
+    impl TDestruct: Destruct<T>,
+    impl TWrappingAdd: WrappingAdd<T>
+>(
+    lhs: u128, rhs: u128
+) -> u128 {
+    // Make sure the value passed in the u128 arguments can fit in the concrete type.
+    let lhs = assert_fits_in_type::<T>(lhs);
+    let rhs = assert_fits_in_type::<T>(rhs);
```

Comment:
Convention says that `into` is not fallible, so no. `into` alone is not a good name. Definitely `expect_into` or something similar that encompasses the fact that it may panic.